### DISCUSSION TITLE
html tag parse

### DISCUSSION THREAD

**Discussion by niuzhi** [fomantic/Fomantic-UI#2367] [Elements (button, image, label, loader, ...)] [answered]
How to use semantic to parse html tag string？

![image](https://user-images.githubusercontent.com/33443724/167539342-8b4a747c-3eb6-4f47-abc9-e6a149ec646a.png)

**Comment by lubber-de** [comment, marked as answer]
I suggest three options

1. JQuery
This will return a Nodelist
```javascript
const domNodes = $.parseHTML($('#parseme')[0].childNodes);
```


2. jQuery parseHTML https://api.jquery.com/jquery.parsehtml/
This will return an array.
```javascript
const domNodes = $.parseHTML($('#parseme').html()));
```

3. Javascript DOMParser https://developer.mozilla.org/en-US/docs/Web/API/DOMParser
This will return a XML Document
```javascript
const domString = document.getElementById('parseme').innerHTML;
const parser = new DOMParser();
const domNodes = parser.parseFromString(domString,'text/html');
```


See all variants in action here (watch console) https://jsfiddle.net/lubber/zfybs7vx/
